packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  FoldedCollection,
+  FoldedResult,
+  FoldedResultList,
+  FoldedResultListState,
+  Result,
+  ResultList,
+  ResultListState,
+  ResultsPerPageState,
+  ResultTemplatesManager,
+  ResultTemplate,
+  buildResultTemplatesManager,
+  ResultListProps,
+  EcommerceDefaultFieldsToInclude,
+} from '@coveo/headless';
+import {identity} from 'lodash';
+import {Bindings} from '../../utils/initialization-utils';
+import {
+  ResultDisplayDensity,
+  ResultDisplayImageSize,
+  ResultDisplayLayout,
+} from '../atomic-result/atomic-result-display-options';
+import {TemplateContent} from '../result-templates/result-template-common';
+
+interface DisplayProps {
+  density: ResultDisplayDensity;
+  imageSize?: ResultDisplayImageSize;
+  image?: ResultDisplayImageSize;
+  display?: ResultDisplayLayout;
+}
+export interface ResultPlaceholderProps extends DisplayProps {
+  resultsPerPageState: ResultsPerPageState;
+}
+export interface ResultsProps extends DisplayProps {
+  resultListState: FoldedResultListState | ResultListState;
+  bindings: Bindings;
+  host: HTMLElement;
+  getContentOfResultTemplate(
+    result: Result | FoldedResult
+  ): HTMLElement | DocumentFragment;
+
+  classes: string;
+}
+
+export interface AtomicResultListBase extends Omit<ResultsProps, 'classes'> {
```

Comment:
Instead of having `AtomicResultListBase`, `ResultsProps`, etc. I think `ResultListCommon` should expose all these attributes & functions somehow. I would prefer not to have both a base interface and a base class, essentially.

Reply:
I've done everything else you guys suggested but I'm not sure how to do this. `AtomicResultListBase` has a few things that need to be in the component itself (and not in the common class), like `bindings`,  `State`s, etc. 
do you have a suggestion of how to achieve this?

Reply:
Maybe `ResultListCommon` doesn't need to be a separate component. `AtomicResultListBase` could perhaps have a render method?

Reply:
@ThibodeauJF done, what do you think?

Reply:
It's much better! I still find some things a bit confusing:

1. `AtomicResultListBase` contains `resultListCommon: ResultListCommon;` meanwhile `ResultListCommon` has the method `public renderList(props: AtomicResultListBase)`. It's a circular reference

2. There is already common props for  `ResultListCommon`, `ResultListCommonProps`, so the `props: AtomicResultListBase` are also common props?

3. `AtomicResultListBase` starts off by extending `ResultsProps` (`AtomicResultListBase extends Omit<ResultsProps, 'classes'>`) which is another common props interface?

Maybe the naming is problematic for me 🤔  I would keep everything with Prop associated to the user facing `Prop()` in Stencil. For the "jsx" props or just common properties/options/params, we can remove the Props and use Options, or something similar